packages/repo/src/car.ts

```diff
@@ -143,18 +152,26 @@ const readCarBlocksIter = (
 }
 
 async function* readCarBlocksIterGenerator(
-  reader: BufferedReader,
+  reader: BytesReader,
 ): AsyncGenerator<CarBlock, void, unknown> {
+  let blocks = 0
   try {
     while (!reader.isDone) {
-      const blockSize = await reader.readVarint()
+      const blockSize = await readVarint(reader)
       if (blockSize === null) {
         break
       }
       const blockBytes = await reader.read(blockSize)
       const cid = parseCidFromBytes(blockBytes.subarray(0, 36))
       const bytes = blockBytes.subarray(36)
       yield { cid, bytes }
+
+      // yield to the event loop every 25 blocks
+      // in the case the incoming CAR is synchronous, this can end up jamming up the thread
+      blocks++
+      if (blocks % 25 === 0) {
+        await new Promise((resolve) => setImmediate(resolve))
```

Comment:
Can pull this right out of node if you want.
```ts
import { setImmediate } from 'node:timers/promises';
// ...
await setImmediate()
```

Reply:
You can even do this, which has the same effect, but might make it even more clear what the intent is:

```ts
import { scheduler } from 'node:timers/promises';
// ...
await scheduler.yield()
```

Reply:
Good call out, though I expect we'll want to hold off while it remains experimental.